libcst/matchers/__init__.py

```diff
@@ -4341,6 +4341,142 @@ class ExceptHandler(BaseMatcherNode):
     ] = DoNotCare()
 
 
+@dataclass(frozen=True, eq=False, unsafe_hash=False)
```

Comment:
Are we worried about the matcher code? I have been putting off trying to understand how matchers work in detail (this code makes zero sense to me without reading through those docs) I can read up on it if you would like.

Reply:
Update: I went ahead and read the matcher docs and I think this makes sense to me, I guess it's generated code and therefore probably correct.

I don't quite follow the OneOf / AllOf business, seems a bit strange to me but that shouldn't block landing